### DISCUSSION TITLE
Connection Refused Error for Chat Service on Port 8065

### DISCUSSION THREAD

**Discussion by dh4340** [cmu-sei/GHOSTS#408] [Q&A] [answered]
Hi,

I'm encountering an issue with the chat service in GHOSTS. The service is attempting to post to port 8065, but I am getting a `Connection refused` error. Here's what I've observed:

### Observations:

1. **Error Details:**

   ```
   2024-09-09 10:39:10.0093|ERROR|ghosts.api.Infrastructure.Animations.AnimationDefinitions.Chat.ChatClient|Cannot login admin@scotty.com:scotty@1 with error Connection refused (localhost:8065)...
   ```

   The error suggests that the chat service is trying to access `http://localhost:8065/api/v4/users/login` but is unable to connect.

2. **Configuration Files:**

   - `docs/animator/run.md` contains:
     ```json
     "PostUrl": "http://localhost:8065"
     ```

   - `src/Ghosts.Api/config/chat.json` includes:
     ```json
     {
       "chat": {
         "baseUrl": "http://localhost:8065/",
         "adminUsername": "admin@scotty.com",
         "adminPassword": "scotty@1"
       }
     }
     ```

   - `src/Ghosts.Api/appsettings.json` has:
     ```json
     "PostUrl": "http://localhost:8065"
     ```

### Issue:

The service is unable to establish a connection to `localhost:8065`, resulting in `Connection refused` errors. The port 8065 is configured in several places, but it's unclear what specific service or application should be running on this port.

### Question:

1. What service is expected to be running on port 8065?
2. How can I resolve the `Connection refused` error for the chat service?
3. Are there any additional configurations needed to ensure the service on port 8065 is accessible?

Any guidance or suggestions on how to troubleshoot this issue would be greatly appreciated!

Thank you!

**Comment by dh4340** [comment]
Figured this one out, but you don't make this clear. I assume from code that you utilize mattermost and run it using the api for chat to work. A sample of this or suggested workflow might assist with preventing confusion.

**Comment by sei-dupdyke** [comment, marked as answer]
Guilty. This is very early work, and so yes, docs are probably quite lacking. We're hoping to get this cleaned up a bit over the fall season. Any feedback here though is greatly appreciated.

**Reply by dh4340** [reply]
You happy for me to create a pull request for docs/code changes? What's the best email address to contact you on as I would like to get this fully working and direct emails would probably help?

**Reply by sei-dupdyke** [reply]
Sure, thank you! It's here: https://github.com/cmu-sei/GHOSTS/blob/master/src/Ghosts.Api/ghosts.api.csproj#L13
